src/rpc/blockchain.cpp

```diff
@@ -406,6 +409,23 @@ void entryToJSON(UniValue &info, const CTxMemPoolEntry &e)
     }
 
     info.pushKV("depends", depends);
+
+    std::set<uint256> set_spent;
+    UniValue spent(UniValue::VARR);
+    for (unsigned int i = 0; i < tx.vout.size(); i++) {
+        const auto& it = mempool.mapNextTx.find(COutPoint(tx.GetHash(), i));
+        if (it == mempool.mapNextTx.end()) {
+            continue;
+        }
+        const uint256& tx_hash = it->second->GetHash();
+        if (mempool.exists(tx_hash)) {
```

Comment:
Is this test really necessary? If `mapNextTx` contains the outpoint, then the spending transaction should be in the mempool? See `CTxMemPool::addUnchecked()`.

Therefore, maybe:
 - replace with `assert(mempool.exists(tx_hash))`
 - or test and throw an error if not in the mempool (or log something)
 - or even remove the check.

IMO not being in `mempool.mapTx` means that something is broken so I would choose the assert.

cc @morcos @sdaftuar

Reply:
I agree. This was a paranoia check and I debated whether this should be an assert. I think I will update as such.